backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/AccumulatingSourceExtractor.java

```diff
@@ -87,7 +87,7 @@ else if ( fieldValue.isJsonNull() ) {
 			// Present, but null
 			return accumulator.accumulate( accumulated, extract( projectionHitMapper, hit, fieldValue, context ) );
 		}
-		else if ( fieldValue.isJsonArray() ) {
+		else if ( !canDecodeArrays() && fieldValue.isJsonArray() ) {
```

Comment:
Note this change, it is isolated in its own commit. problem here was that accumulator would start feeding vector elements to the codec that expects to get an entire array...

Reply:
This will probably lead to problems with arrays of arrays, but since vector fields cannot currently be multi-valued... +1 let's do this.